debezium-core/src/test/java/io/debezium/relational/TableSchemaBuilderTest.java

```diff
@@ -297,17 +298,60 @@ public void shouldSanitizeFieldNamesAndBuildTableSchemaFromTableWithoutPrimaryKe
         assertThat(value.get("C5")).isEqualTo(ByteBuffer.wrap(new byte[]{ 71, 117, 110, 110, 97, 114}));
         assertThat(value.get("C6")).isEqualTo(Short.valueOf((short) 0));
     }
-
+    
     @Test
     @FixFor("DBZ-1044")
     public void shouldSanitizeFieldNamesAndValidateSerialization() {
         schema = new TableSchemaBuilder(new JdbcValueConverters(), adjuster, SchemaBuilder.struct().build(), true)
-                .create(prefix, "sometopic", table, null, null);
+                .create(prefix, "sometopic", table, null, null, null);
 
         Struct key = (Struct) schema.keyFromColumnData(keyData);
         Struct value = schema.valueFromColumnData(data);
 
         SourceRecord record = new SourceRecord(Collections.emptyMap(), Collections.emptyMap(), "sometopic", schema.keySchema(), key, schema.valueSchema(), value);
         VerifyRecord.isValid(record);
     }
+
+    @Test
+    @FixFor("DBZ-1015")
+    public void shouldBuildTableSchemaFromTableWithCustomKey() {
+        table = table.edit().setPrimaryKeyNames().create();
+        schema = new TableSchemaBuilder(new JdbcValueConverters(), adjuster, SchemaBuilder.struct().build(), false)
+                .create(prefix, "sometopic", table, null, null, CustomKeyMapper.getInstance("(.*).table:C2,C3"));
+        assertThat(schema).isNotNull();
+        Schema keys = schema.keySchema();
+        assertThat(keys).isNotNull();
+        assertThat(keys.fields()).hasSize(2);
+        assertThat(keys.field("C2").name()).isEqualTo("C2");
+        assertThat(keys.field("C3").name()).isEqualTo("C3");
+    }
+    
+    @Test
+    @FixFor("DBZ-1015")
+    public void shouldOverrideIdentityKey() {
+        schema = new TableSchemaBuilder(new JdbcValueConverters(), adjuster, SchemaBuilder.struct().build(), false)
+                .create(prefix, "sometopic", table, null, null, CustomKeyMapper.getInstance("(.*).table:C2,C3"));
+        assertThat(schema).isNotNull();
+        Schema keys = schema.keySchema();
+        assertThat(keys).isNotNull();
+        assertThat(keys.fields()).hasSize(2);
+        assertThat(keys.field("C1")).isNull();
+        assertThat(keys.field("C2").name()).isEqualTo("C2");
+        assertThat(keys.field("C3").name()).isEqualTo("C3");
+    }
+    
+    @FixFor("DBZ-1015")
+    public void shouldFallbackToIdentyKeyWhenCustomMapperIsInvalid() {
```

Comment:
What does "invalid" mean exactly? If the given PK configuration doesn't make sense, I'd suggest to rather fail early and raise a validation error.

Reply:
Validation added to the Field (regex)